ddtrace/internal/rate_limiter.py

```diff
@@ -0,0 +1,79 @@
+import time
+import threading
+
+
+class RateLimiter(object):
+    """
+    A token bucket rate limiter implementation
+    """
+    __slots__ = ('rate_limit', 'tokens', 'max_tokens', 'last_update', '_lock')
+
+    def __init__(self, rate_limit):
+        """
+        Constructor for RateLimiter
+
+        :param rate_limit: The rate limit to apply for number of transactions per second.
```

Comment:
what's a transaction in this context? I never encountered the term before

Reply:
I wanted to go for a generic enough term since this rate limiter isn't tied to anything specific (e.g. we could reuse it for other rate limiting for something not related to sampling).

Open to other suggestions, "transactions" was the best I could come up with in the moment 😆